### ISSUE TITLE
[API Bug] Typing Korean characters in `Form.TextArea` shows unexpected behavior

### ISSUE THREAD

**Issue by jcha0713** [raycast/extensions#2646]
OS: MacOS Monterey 12.3
Raycast version: 1.39.1

## Description

When typing in Korean(Hangul) on `Form.TextArea`, the text area does not seem to take all the inputs I type. For example, 

https://user-images.githubusercontent.com/29053796/185813006-2396069b-431b-4cb5-a8d5-501cf806c70f.mp4

in this video, I typed `안녕하세요` twice, but the corresponding output is either `안녕ㅏㅔㅛ` or `안ㅕ하요` which clearly does not match with the word I typed.

However, when I get rid of the `onChange` part from the `From.TextArea` component and prevent it from setting a new state when I type, it works just fine:

https://user-images.githubusercontent.com/29053796/185813250-a3f1a90f-b90a-4809-9772-76b6071320c9.mp4

This time it shows `안녕하세요 안녕하세요`. (this is correct)
The described behavior is also present when using the `Form.TextField` component.

## Steps To Reproduce

1. create an extension with the `Create Extension` command in Raycast
2. Go to the `.tsx` file (or create one if you don't have one) and add the following

```jsx
import { useState } from "react";
import { Form } from "@raycast/api";

export default function Command() {
  const [description, setDescription] = useState<string>();

  return (
    <Form>
      <Form.TextArea id="description" value={description} onChange={setDescription} />
    </Form>
  );
}
```

4. run `npm install && npm run dev` and type in Korean into the text area.

## The current behavior

The `Form.TextArea` does not behave correctly when typing in Korean.

## The expected behavior

It should take all the inputs I type and display them correctly.

**Comment by mathieudutour** [comment]
Hey, I'm looking into this.

Does this happens when writing in a `TextField` as well or only `TextArea`?

I don't understand Korean so I don't know what key to press to reproduce this behaviour. Could you write down what exact keys to press after I switched my keyboard input?

**Comment by jcha0713** [comment]
Thank you for reviewing this issue.

Yes, the problem persists even when I changed `TextArea` into `TextField`.
Once you switched your keyboard input to `2-set Korean`(the standard Korean layout), and press `dkssudgktpdy`, you should see `안녕하세요` in the text area.

Tested on 1.41.0 with the following code from the api documentation.

```JSX
import { ActionPanel, Form, Action } from "@raycast/api";
import { useState } from "react";

export default function Command() {
  const [description, setDescription] = useState<string>("");

  return (
    <Form
      actions={
        <ActionPanel>
          <Action.SubmitForm title="Submit Description" onSubmit={(values) => console.log(values)} />
        </ActionPanel>
      }
    >
      <Form.TextArea id="description" value={description} onChange={setDescription} />
    </Form>
  );
}
```

**Comment by mathieudutour** [comment]
Should be fixed in 1.45!
